rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystem.java

```diff
@@ -306,22 +248,36 @@ else if (start >= 0L && len >= 0L) {
       else {
         // Using similar logic of returning all Blocks as FileSystem.getFileBlockLocations does instead of only returning blocks from start till len
 
-        BlockLocation[] blockLocations = new BlockLocation[(int) Math.ceil((double) file.getLen() / clusterManager.getSplitSize())];
-        int blockNumber = 0;
-        for (long i = 0; i < file.getLen(); i = i + clusterManager.getSplitSize()) {
-          long end = i + clusterManager.getSplitSize();
-          if (end > file.getLen()) {
-            end = file.getLen();
+        try {
+          BlockLocation[] blockLocations = new BlockLocation[(int) Math.ceil((double) file.getLen() / splitSize)];
+          int blockNumber = 0;
+
+          RetryingBookkeeperClient client = new BookKeeperFactory().createBookKeeperClient(conf);
+
+          for (long i = 0; i < file.getLen(); i = i + splitSize) {
+            long end = i + splitSize;
+            if (end > file.getLen()) {
+              end = file.getLen();
+            }
+            String key = file.getPath().toString() + i + end;
+            String hostName = client.getClusterNodeHostName(key, clusterType.ordinal());
```

Comment:
any impact on query performance after this change? We have added more network calls in critical sections after this change.

Reply:
We haven't measured the performance yet but this is not over the network (cross node every time). Most of the time getFileBlockLocations gets invoked in the coordinator and in that case its a call to bookkeeper daemon running on the same node. For the case where this code is running in the Worker node, it will be a call to the coordinator only when the cache has expired.